collectd/src/main/java/com/microsoft/applicationinsights/collectd/ApplicationInsightsWriter.java

```diff
@@ -0,0 +1,227 @@
+
+    // endregion Members
+
+    // region Ctors
+
+    public ApplicationInsightsWriter() {
+        Collectd.registerConfig(PLUGIN_NAME, this);
+        Collectd.registerInit(PLUGIN_NAME, this);
+        Collectd.registerWrite(PLUGIN_NAME, this);
+        Collectd.registerShutdown(PLUGIN_NAME, this);
+
+        this.logger = new ApplicationInsightsWriterLogger();
+        this.telemetryConfiguration = TelemetryConfiguration.getActive();
+    }
+
+    protected ApplicationInsightsWriter(TelemetryClient telemetryClient, ApplicationInsightsWriterLogger logger) {
+        this.telemetryClient = telemetryClient;
+        this.telemetryConfiguration = TelemetryConfiguration.getActive();
+        this.logger = logger;
+    }
+
+    // endregion Ctors
+
+    // region Public
+
+    /**
+     * Reads the configuration provided for the writer plugin.
+     * @param configuration The configuration.
+     * @return Zero on success, false otherwise.
+     */
+    public int config(OConfigItem configuration) {
+        logger.logInfo("Loading configuration.");
+        logger.logDebug("Configuration found: " + configuration);
+
+        List<OConfigItem> children = configuration.getChildren();
+        if (children.size() < 1) {
+            logger.logError("Configuration error: application instrumentation key must be set.");
+
+            return 1;
+        }
+
+        for (int i = 0; i < children.size(); i++) {
+            OConfigItem child = children.get(i);
+            String key = child.getKey();
+
+            if (key.equalsIgnoreCase(INSTRUMENTATION_KEY_CONFIGURATION_KEY)) {
+                List<OConfigValue> values = child.getValues();
+                if (values.size() != 1) {
+                    logger.logError(key + " configuration option needs exactly 1 argument.");
+                    return 2;
+                }
+
+                String instrumentationKey = values.get(0).toString();
+                if (LocalStringsUtils.isNullOrEmpty(instrumentationKey)) {
+                    logger.logError (INSTRUMENTATION_KEY_CONFIGURATION_KEY + "' configuration option is mandatory, plugin will be disabled");
+
+                    return 1;
+                }
+
+                this.telemetryConfiguration.setInstrumentationKey(instrumentationKey);
+            } else if (key.equalsIgnoreCase(SDK_LOGGER_CONFIGURATION_KEY)) {
+                List<OConfigValue> values = child.getValues();
+                if (values.size() != 1) {
+                    logger.logError(key + " configuration option needs exactly 1 argument [true/false].");
+                    return 2;
+                }
+
+                boolean enableSDKLogger = Boolean.parseBoolean(values.get(0).toString());
+                if (enableSDKLogger) {
+                    InternalLogger.INSTANCE.initialize("CONSOLE", new HashMap<String, String>());
+                }
+            } else {
+                logger.logWarning("Unknown configuration option '" + key + "'.");
```

Comment:
Is it possible that this configuration key is intended to another writer or to CollectD core?
I'm asking because in the case above this shouldn't be a warning.